boa/src/builtins/value/display.rs

```diff
@@ -0,0 +1,225 @@
+use super::*;
+
+impl Display for Value {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        Display::fmt(&self.data(), f)
+    }
+}
+
+/// A helper macro for printing objects
+/// Can be used to print both properties and internal slots
+/// All of the overloads take:
+/// - The object to be printed
+/// - The function with which to print
+/// - The indentation for the current level (for nested objects)
+/// - A HashSet with the addresses of the already printed objects for the current branch
+///      (used to avoid infinite loops when there are cyclic deps)
+macro_rules! print_obj_value {
+    (all of $obj:expr, $display_fn:ident, $indent:expr, $encounters:expr) => {
+        {
+            let mut internals = print_obj_value!(internals of $obj, $display_fn, $indent, $encounters);
+            let mut props = print_obj_value!(props of $obj, $display_fn, $indent, $encounters, true);
+
+            props.reserve(internals.len());
+
+            props.append(&mut internals);
+
+            props
+        }
+    };
+    (internals of $obj:expr, $display_fn:ident, $indent:expr, $encounters:expr) => {
+        print_obj_value!(impl internal_slots, $obj, |(key, val)| {
+            format!(
+                "{}{}: {}",
+                String::from_utf8(vec![b' '; $indent])
+                                .expect("Could not create indentation string"),
```

Comment:
I would use [`str::repeat`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.repeat) here. It should remove the `expect()` branch, the UTF-8 checking, and we could personalize the indentation by either using 2 spaces, 4 spaces or a tab, with a constant (or multiple constants).

Reply:
I think we should use `fromat!("{:>width}: {}", key, ..., width = $ident)` this eliminates the unnecessary string construction